### PR TITLE
NIB Additions to Table and Collection view cells

### PR DESCRIPTION
**Pull request by beloso**
I have decided to revive @pbaltarejo 's PR.

I am aware that this requires some Unit Tests. What would you think about adding nibs / xibs to the test target and then we could test using those xibs / nibs. Sorry if this sounds silly, I am no text expert, I am looking for some advice on how to test this.

### CODE REVIEW

File: Sources/View/ReusableView.swift [hunk 1]
```diff
@@ -65,7 +80,8 @@ public extension UICollectionView {
 
     @available(iOS 9, *)
```
Comment (p4checo) [inline Sources/View/ReusableView.swift]: I think we can probably remove this, as the project has had iOS 9 as deployment target for a while now 👴 (same on .podspec)

File: Sources/View/ReusableView.swift [hunk 2]
```diff
@@ -17,28 +17,25 @@ extension UITableViewHeaderFooterView: ReusableView {}
...
-    func register<T: UICollectionViewCell>(_ cellType: T.Type)
-    where T: ReusableView {
+    func register<T: UICollectionViewCell>(_ cellType: T.Type) {
         register(cellType, forCellWithReuseIdentifier: cellType.reuseIdentifier)
     }
 
     func register<T: UICollectionViewCell>(_ cellType: T.Type)
-    where T: ReusableView, T: NibView {
-        register(cellType.nib, forCellWithReuseIdentifier: cellType.reuseIdentifier)
+        where T: NibView {
```
Comment (p4checo) [inline Sources/View/ReusableView.swift]: incorrect indentation ( more cases below) 

also, please move this to the same line, or alternatively combine on the  generic parameter, like `<T: UICollectionViewCell & NibView>` (we haven't settled on a chosen variant in our code style yet)

**Comment by belosoGrocerKey** [comment]
I will add unit tests next week. Have a busy weekend planned ahead

File: Tests/AlicerceTests/View/Nibs/TestNIBCollectionReusableView.swift [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// Copyright © 2019 Mindera. All rights reserved.
```
Comment (p4checo) [inline Tests/AlicerceTests/View/Nibs/TestNIBCollectionReusableView.swift]: please remove these headers (more cases below)

File: Tests/AlicerceTests/View/Nibs/TestNIBCollectionReusableView.swift [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+import UIKit
+
+class TestNIBCollectionReusableView: CollectionReusableView, NibView {
+	@IBOutlet weak var testLabel: UILabel!
```
Comment (p4checo) [inline Tests/AlicerceTests/View/Nibs/TestNIBCollectionReusableView.swift]: we use 4 spaces, and not tabs I think 🚓

File: Tests/AlicerceTests/View/ReusableViewCollectionViewTestCase.swift [hunk 5]
```diff
@@ -26,65 +26,96 @@ class ReusableViewCollectionViewTestCase: XCTestCase {
 
     func testRegister_WithCollectionViewCell_ShouldSucceedl() {
         collectionView.register(TestCollectionViewCell.self)
+		collectionView.register(TestNIBCollectionViewCell.self)
```
Comment (p4checo) [inline Tests/AlicerceTests/View/ReusableViewCollectionViewTestCase.swift]: please clean up all these tabs... 👮

**Pull request merged**